lib/Dialect/TritonGPU/Transforms/ReorderInstructions.cpp

```diff
@@ -117,6 +117,8 @@ class TritonGPUReorderInstructionsPass
       auto dotUser = dyn_cast<triton::DotOp>(*op->user_begin());
       if (!dotUser)
         return;
+      if (std::distance(op->user_begin(), op->user_end()) > 1)
+        return;
```

Comment:
can you merge this with the condition `if (op->getUsers().empty())` to become  `llvm::hasSingleElement(op->getUsers())`

Reply:
done

Reply:
We should call `hasOneUse()`, no?